experimental/devices/as7262/as7262.go

```diff
@@ -221,16 +233,18 @@ func (d *Dev) Sense(ledDrive physic.ElectricCurrent, senseTime time.Duration) (S
 	}, nil
 }
 
-var waitForSensor = time.After
-
 // Halt stops any pending operations. Repeated calls to Halt do nothing.
 func (d *Dev) Halt() error {
-	d.canceledMu.Lock()
-	defer d.canceledMu.Unlock()
-
-	if !d.canceled {
-		d.done <- struct{}{}
-		d.canceled = true
+	d.cancelMu.Lock()
+	defer d.cancelMu.Unlock()
+	d.cancel()
+	select {
```

Comment:
There's no need for select now, just do:
```
_, _ = <- d.done
```

and that's it.

Reply:
That's much cleaner.